src/analyses/useAfterFree.ml

```diff
@@ -0,0 +1,188 @@
+    let filtered_args = List.filter (fun x -> check_exp x ctx) args in
+    let args_to_add_to_callee_state = List.map (fun x ->
+        match ctx.ask (Queries.EvalValue x) with
+        | a when not (Queries.VD.is_top a) ->
+          begin match a with
+            | `Blob (v, s, t) -> (v, s, t)
+            | _ -> ValueDomain.Blobs.top () (* TODO: Is this correct? *)
+          end
+        | _ -> ValueDomain.Blobs.top () (* TODO: Is this correct? *)
+      ) filtered_args in
+    let callee_state = D.of_list args_to_add_to_callee_state in
+    match lval with
+    | Some lval ->
+      if check_lval lval ctx then
+        M.warn "WARN: lval in enter contains maybe freed blob";
+      [caller_state, callee_state]
+    | None -> [caller_state, callee_state]
+
+  (* TODO: Unsure about this transfer function. Should we leave it as an identity function? *)
+  let combine_env ctx (lval:lval option) fexp (f:fundec) (args:exp list) fc (callee_local:D.t) (f_ask:Queries.ask) : D.t =
+    callee_local
+
+  let combine_assign ctx (lval:lval option) fexp (f:fundec) (args:exp list) fc (callee_local:D.t) (f_ask: Queries.ask): D.t =
+    let state = ctx.local in
+    let () = List.iter (fun x ->
+        if check_exp x ctx then
+          M.warn "argument in combine_assign contains maybe freed blob"
+      ) args in
+    match lval with
+    | Some lval ->
+      if check_lval lval ctx then
+        M.warn "lval in combine_assign contains maybe freed blob";
+      state
+    | None -> state
+
+  let special ctx (lval:lval option) (f:varinfo) (arglist:exp list) : D.t =
+    let state = ctx.local in
+    let desc = LibraryFunctions.find f in
+    match desc.special arglist with
+    | Free ptr ->
+      if not (check_exp ptr ctx) then
+        begin match ctx.ask (Queries.EvalValue ptr) with
+          | a when not (Queries.VD.is_top a) ->
+            begin match a with
+              | `Blob (v, s, t) -> D.add (v, s, t) state
+              | _ -> state
+            end
+          | _ -> state
+        end
+      else
+        (* TODO: Is this way of printing in the else block okayish? *)
+        let () = M.warn "WARN: trying to free an already freed ptr" in
+        state
+    (*
+     * Check if we're allocating a maybe freed ptr -> in this case remove
+     * it from the maybe freed set
+     * TODO: Does this make sense?
+     *)
+    | Malloc _
+    | Calloc _
+    | Realloc _ ->
+      begin match lval with
+        | Some lval ->
+          begin match lval with
+            | (Mem e, _) ->
+              begin match ctx.ask (Queries.EvalValue e) with
+                | a when not (Queries.VD.is_top a) ->
+                  begin match a with
+                    | `Blob (v, s, t) ->
+                      (*
+                       * If we're trying to allocate a blob that is maybe freed,
+                       * then we can remove it from the maybe freed set
+                       * TODO: Makes sense?
+                       *)
```

Comment:
I'm not sure if the corresponding blob is even already created here, so I fear this will usually yield `VD.top ()`. Also, upon newly creating the blob here, the abstract value inside it will usually not be the same as the one when free was called. So `mem` will probably never trigger.